### PR TITLE
Transfer files - moving errors handling to new phase

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
refactor error retry from phase 2 into a new phase 3.
moving this phase to be the last operation, adding a progress bar to show the progress and count of uploading errors.

### CODE REVIEW

**Comment by attiasas** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by attiasas** [comment]
retry

**Comment by attiasas** [comment]
recheck

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'd like to review the PR again, following the implementation of the below throughout the code.
* Our convention is to start code comments with an upper case letter. For example, let's change - "// reads an error..." into "// Reads an error..."
* Let's make sure the keep correct grammar when adding comments. For example, notice the following comments. Since it starts with "Reads", it should then include "parses" and not "parse" - 
```
// reads an error file from a given path, parse and populate a given FilesErrors instance with the file information
```

File: artifactory/commands/transferfiles/fileserror.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	if err != nil {
+		return true, err
+	}
+	return len(e.errorsFilesToHandle) < 1, nil
+}
+
+func (e *errorsRetryPhase) phaseStarted() error {
+	e.startTime = time.Now()
+	return nil
+}
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/fileserror.go:94]: The above `phaseStarted` code is good, but let's simplify a bit the code of the `phaseStarted` functions of the other two pahses.
The following - 
```
func (f *filesDiffPhase) phaseStarted() error {
	f.startTime = time.Now()
	err := addNewDiffToState(f.repoKey, f.startTime)
	if err != nil {
		return err
	}

	return err
}
```
can be simplified to - 
```
func (f *filesDiffPhase) phaseStarted() error {
	f.startTime = time.Now()
	return addNewDiffToState(f.repoKey, f.startTime)
}
```
Let's make a similar simplification in last `phaseStarted` function.

File: artifactory/commands/transferfiles/fileserror.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+func (e *errorsRetryPhase) initProgressBar() error {
+	if e.progressBar == nil {
+		return nil
+	}
+
+	// Init progress with the number of tasks of errors file handling (fixing previous upload failures)
+	filesCount := 0
+	for _, path := range e.errorsFilesToHandle {
+		//var failedFiles FilesErrors
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/fileserror.go:104]: Let's remove this commenrt. It's unclear.

File: artifactory/commands/transferfiles/fileserror.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		failedFiles, err := readErrorFile(path)
+		if err != nil {
+			return err
+		}
+		filesCount += len(failedFiles.Errors)
+	}
+
+	log.Debug("Starting Error-Retry Phase (3) for '", filesCount, "' files")
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/fileserror.go:112]: This debug line seems unnecessary.

File: artifactory/commands/transferfiles/transferfileprogress.go [hunk 4]
```diff
@@ -170,6 +170,10 @@ func (t *TransferProgressMng) AddPhase2(tasksPhase2 int64) {
 	t.phases = append(t.phases, t.barsMng.NewTasksWithHeadlineProg(tasksPhase2, "Phase 2: Transferring newly created and modified files", false, progressbar.GREEN, TimeSlots.String()))
 }
 
+func (t *TransferProgressMng) AddPhase3(tasksPhase3 int64) {
+	t.phases = append(t.phases, t.barsMng.NewTasksWithHeadlineProg(tasksPhase3, "Phase 3: Retry transferring all files that had upload errors", false, progressbar.GREEN, Files.String()))
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/transferfileprogress.go:174]: Retry transferring all files that had upload errors
-->
Retry transfer failures

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**